quesma/queryparser/pancake_aggregation_parser_buckets.go

```diff
@@ -359,6 +360,35 @@ func (cw *ClickhouseQueryTranslator) parseComposite(aggregation *pancakeAggregat
 	return nil
 }
 
+func (cw *ClickhouseQueryTranslator) parseIpPrefix(aggregation *pancakeAggregationTreeNode, params QueryMap) error {
+	const (
+		defaultIsIpv6             = false
+		defaultAppendPrefixLength = false
+		defaultKeyed              = false
+		defaultMinDocCount        = 1
+	)
+
+	if err := bucket_aggregations.CheckParamsIpPrefix(cw.Ctx, params); err != nil {
+		return err
+	}
+
+	aggr := bucket_aggregations.NewIpPrefix(
+		cw.Ctx,
+		cw.parseFieldField(params, "ip_prefix"),
+		cw.parseIntField(params, "prefix_length", 0), // default doesn't matter, it's required
+		cw.parseBoolField(params, "is_ipv6", defaultIsIpv6),
+		cw.parseBoolField(params, "append_prefix_length", defaultAppendPrefixLength),
+		cw.parseBoolField(params, "keyed", defaultKeyed),
+		cw.parseIntField(params, "min_doc_count", defaultMinDocCount),
+	)
```

Comment:
Looks nice, thinking about generic representation:
https://www.geeksforgeeks.org/reflect-tag-get-function-in-golang-with-examples/